File: app/src/main/java/net/north101/android/ghplayertracker/PerkNoteAdapter.java
package net.north101.android.ghplayertracker;

import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import net.north101.android.ghplayertracker.data.PerkNote;

public class PerkNoteAdapter extends RecyclerView.Adapter<PerkNoteViewHolder> {
    public final PerkNote[] items;

    public PerkNoteAdapter(PerkNote[] items) {
        this.items = items;
    }

    @Override
    public PerkNoteViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return PerkNoteViewHolder.inflate(parent);
    }

    @Override
    public void onBindViewHolder(PerkNoteViewHolder holder, int position) {
        holder.bind(items[position]);
    }

    @Override
    public int getItemCount() {
        return items.length;
    }
}

File: app/src/main/java/net/north101/android/ghplayertracker/PerkNoteAdapter.kt
package net.north101.android.ghplayertracker

import android.support.v7.widget.RecyclerView
import android.view.ViewGroup

class PerkNoteAdapter : RecyclerView.Adapter<CharacterPerkNoteViewHolder>() {
    val items = ArrayList<PerkNoteData>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CharacterPerkNoteViewHolder {
        return CharacterPerkNoteViewHolder.inflate(parent)
    }

    override fun onBindViewHolder(holder: CharacterPerkNoteViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int {
        return items.size
    }

    fun updateItems(items: List<PerkNoteData>) {
        this.items.clear()
        this.items.addAll(items)
        this.notifyDataSetChanged()
    }
}
